kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -128,6 +147,55 @@ pub fn extract_harness_attributes(tcx: TyCtxt, def_id: DefId) -> Option<HarnessA
     }
 }
 
+/// Check that any unstable API has been enabled. Otherwise, emit an error.
+/// TODO: Improve error message by printing the span of the callee instead of the definition.
+pub fn check_unstable_features(tcx: TyCtxt, enabled_features: &[String], def_id: DefId) {
+    let attributes = extract_kani_attributes(tcx, def_id);
+    if let Some(unstable_attrs) = attributes.get(&KaniAttributeKind::Unstable) {
+        for attr in unstable_attrs {
+            match UnstableAttribute::try_from(*attr) {
+                Ok(unstable_attr) => {
+                    if !enabled_features.contains(&unstable_attr.feature) {
+                        let fn_name = tcx.def_path_str(def_id);
+                        let reason =
+                            unstable_attr.reason.map_or(String::default(), |r| format!(": {r}"));
+                        let mut err = tcx.sess.struct_err(format!(
+                            "Use of unstable feature `{}`{reason}",
+                            unstable_attr.feature
+                        ));
+                        err.span_note(
+                            tcx.def_span(def_id),
+                            format!("the function `{fn_name}` is unstable:"),
+                        );
+                        if let Some(link) = unstable_attr.issue {
+                            err.note(format!("see issue {link} for more information"));
+                        }
+                        err.help(format!(
+                            "use `-Z {}` to enable use this API.",
+                            unstable_attr.feature
+                        ));
+                        err.emit();
+                    }
+                }
+                Err(msg) => {
```

Comment:
Since this case is meant for developers and not users, you might want to change it to a panic.

Reply:
I thought about it, but my worry is that we don't have anyway of actually restricting the usage of `kani::unstable` in the user code. But I can try to come up with something.

Reply:
I think it's very unlikely for users to use it.

Reply:
For sure. It would be a mistake, but since it is public, I rather treat it as so. What if I add a debug assert?

Reply:
Sure